homeassistant/components/sensor/whois.py

```diff
@@ -121,16 +109,30 @@ def update(self):
             return
 
         if response:
-            self._data = response
+            if 'expiration_date' not in response:
+                _LOGGER.error(
+                    "Failed to find expiration_date in whois lookup response. "
+                    "Did find: %s", ', '.join(response.keys()))
+                return
+
+            if not response['expiration_date']:
+                _LOGGER.error("Whois response expiration_date empty.")
+                return
+
+            expiration_date = response['expiration_date'][0]
+            self._attributes[ATTR_EXPIRES] = expiration_date.isoformat()
```

Comment:
We should start building a new dict here, in case keys that were present once don't get implicitly carried over. (Not likely for WHOIS case, but it's still good practice)
```python
attributes = {
    ATTR_EXPIRES: expiration_date.isoformat(),
}

if 'nameservers' in response:
    attributes[ATTR_NAME_SERVERS] = ' '.join(response['nameservers'])
...
self._attributes = attributes
```

Reply:
Added a local dict which gets populated and overwrite the WhoisSensor `_attributes` attribute with.